python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1618,6 +1618,16 @@ def report_run_failed(self, pipeline_run, message=None):
             "This run has been marked as failed from outside the execution context.",
         )
 
+        run = self.get_run_by_id(pipeline_run.run_id)
+        if run is None:
+            raise DagsterInvariantViolationError(
+                f"Could not load run {pipeline_run.run_id} that was passed to report_run_failed"
+            )
+        if run.status not in IN_PROGRESS_RUN_STATUSES:
```

Comment:
I'm not confident that this same check is appropriate for guarding state transition to failed.  I could see this just checking for success if transitions from other statuses should be allowed.

Reply:
Runs can fail from `NOT_STARTED`, `QUEUED`, and `MANAGED`

Reply:
changed to just recheck during monitoring loop